File: ArrayAndStrings/remove-dup-from-sorted-2/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  const map = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (map.has(nums[i])) {
      map.get(nums[i]).push(i);
    } else {
      map.set(nums[i], [i]);
    }
  }
  let count = 0;
  for (let indexes of map.values()) {
    while (indexes.length > 2) {
      indextoDel = indexes.pop();
      nums[indextoDel] = Infinity;
      count++;
    }
  }
  nums.sort((a, b) => a - b);
  let i = count;
  while (i >= 1) {
    nums.pop();
    i--;
  }
};

File: ArrayAndStrings/remove-dup-from-sorted-2/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        key = dict()
        for i in range(len(nums)):
            if nums[i] in key:
                key[nums[i]].append(i)
            else:
                key[nums[i]] = [i]
        count = 0
        for (key, indexes) in key.items():
            while len(indexes) > 2:
                indexRem = indexes.pop()
                nums[indexRem] = float("inf")
                count +=1
        nums.sort()
        
        i = count
        while i >=1:
            nums.pop()
            i-=1
        